File: data/imported/knowledge/cells/typescript/merged/snippets/snippet_004.ts
async function testMixedValues() {
  const workbook = new Workbook();
  const worksheet = workbook.worksheets[0]!;

  worksheet.putValue("A1", 42);
  worksheet.putValue("A2", 3.14159);
  worksheet.putValue("A3", "Hello World");
  const cellA4 = worksheet.getCell2("A4");
  cellA4.setFormula("=SUM(A1:A2)");
  worksheet.putValue("A5", -100);
  worksheet.putValue("A6", 2.71828);
  worksheet.putValue("A7", "");
  const cellA8 = worksheet.getCell2("A8");
  cellA8.setFormula("=A1+A2");
  worksheet.putValue("A9", "Test String");
  worksheet.putValue("A10", 0);

  await workbook.save("outputfiles/test_mixed_values.xlsx");
  console.log("Saved to outputfiles/test_mixed_values.xlsx");

  const loaded = await Workbook.load("outputfiles/test_mixed_values.xlsx");
  const ws = loaded.worksheets[0]!;
  console.log("Loaded A1:", ws.getCell(0, 0)?.value);
  console.log("Loaded A3:", ws.getCell(2, 0)?.value);
  console.log("Loaded A4 formula:", ws.getCell(3, 0)?.formula);
}

File: data/imported/knowledge/cells/go/merged/snippets/snippet_004.go
func main() {
	wb := cells_foss.NewWorkbook()
	ws := wb.Worksheets[0]

	// Set up header and data cells.
	ws.Cells().Set("A1", "Fruit")
	ws.Cells().Set("A2", "Apple")
	ws.Cells().Set("A3", "Banana")

	// Create a list-type data validation.
	dv := &cells_foss.DataValidation{
		Type:             cells_foss.DataValidationTypeList,
		Formula1:         `"Apple,Banana,Cherry,Dragonfruit"`,
		AllowBlank:       true,
		ShowErrorMessage: true,
		ErrorTitle:       "Invalid Fruit",
		ErrorMessage:     "Please pick a fruit from the list.",
		ErrorStyle:       cells_foss.ErrorStyleStop,
	}

	if err := ws.AddDataValidation("A2:A10", dv); err != nil {
		fmt.Printf("Error: %v\n", err)
		return
	}
	fmt.Printf("Added %q validation on A2:A10\n", dv.Type)

	// Add a second validation — whole number range.
	dv2 := &cells_foss.DataValidation{
		Type:             cells_foss.DataValidationTypeWhole,
		Formula1:         "1",
		Formula2:         "100",
		ShowErrorMessage: true,
		ErrorTitle:       "Invalid Value",
		ErrorMessage:     "Enter a whole number between 1 and 100.",
		ErrorStyle:       cells_foss.ErrorStyleWarning,
	}
	ws.Cells().Set("B1", "Score (1-100)")
	ws.AddDataValidation("B2:B10", dv2)
	fmt.Println("Added whole-number validation on B2:B10 (1-100)")

	outPath := "outputfiles/data_validation.xlsx"
	if err := wb.Save(outPath); err != nil {
		fmt.Printf("Error saving workbook: %v\n", err)
		return
	}
	fmt.Printf("Workbook saved to %s\n", outPath)
	fmt.Println("Open in Excel — A2:A10 shows a dropdown; B2:B10 accepts 1-100.")
}